galley/pkg/config/analysis/analyzers/virtualservice/destinationhosts.go

```diff
@@ -75,13 +78,40 @@ func (d *DestinationHostAnalyzer) analyzeVirtualService(r *resource.Entry, ctx a
 }
 
 func (d *DestinationHostAnalyzer) checkDestinationHost(vsNamespace string, destination *v1alpha3.Destination,
-	ctx analysis.Context, serviceEntryHosts map[resource.Name]bool) bool {
+	ctx analysis.Context, serviceEntryHosts []resource.Name) bool {
 
 	name := util.GetResourceNameFromHost(vsNamespace, destination.GetHost())
-
-	// Check explicitly defined ServiceEntries as well as services discovered from the platform
-	if _, ok := serviceEntryHosts[name]; ok {
-		return true
+	_, cmpHost := name.InterpretAsNamespaceAndName()
+
+	// Check defined ServiceEntries as well as services discovered from the platform
+	for _, seFullHost := range serviceEntryHosts {
```

Comment:
Since this is now a more expensive check, we should probably move it below the check against exact SyntheticServiceEntry matches. 

If we want to be more efficient, then we should keep the non-wildcard service entry hosts in a map and also create a list of wildcard service entry hosts. Then we can do the map checks first for exact matches, and the wildcard checks last.

Reply:
I did consider this, although I wasn't sure which one is really more efficient. Linear search is quite fast with small amount of data and creating the map isn't free either. Although I guess we could go with the separated queries since the cost doesn't increase as fast.